python/pyspark/sql/session.py

```diff
@@ -161,8 +161,8 @@ def getOrCreate(self):
             with self._lock:
                 from pyspark.context import SparkContext
                 from pyspark.conf import SparkConf
-                session = SparkSession._instantiatedContext
-                if session is None:
+                session = SparkSession._instantiatedSession
+                if session is None or session._sc._jsc is None:
```

Comment:
Actually `_jsc` is a `JavaSparkContext` instead of `SparkContext` and it doesn't have `isStopped`.